### ISSUE TITLE
[Lab1] Βήμα 19: Υλοποίηση κλάσης PytorchNNModel

### ISSUE THREAD

**Issue by raniatze** [slp-ntua/patrec-labs#31]
Καλησπέρα,

Θα ήθελα να κάνω μια ερώτηση σχετικά με την υλοποίηση της κλάσης PytorchNNModel στο βήμα 19.  

Για τις μεθόδους fit και predict χρειαζόμαστε όπως φαίνεται και στο lib.py να ορίσουμε τα **train_loader**, **val_loader** (για την fit) και **test_loader** (για την predict). Αυτό προυποθέτει αρχικά τον ορισμό των αντίστοιχων datasets τα οποία θα δοθούν στη συνέχεια στην dataloader που υλοποίησαμε στο βήμα 19 (α). Για την δημιουργία ενός dataset έχουμε φτιάξει σε προηγούμενο πάλι βήμα μία ξεχωριστή κλάση **DigitsDataset(X,y)**. 

Επομένως, πώς θα δημιουργήσουμε τα απαραίτητα objects (datasets) αυτής της κλάσης (DigitsDataset) εντός της PytorchNNModel; 

Επίσης στην predict(X) πώς θα δημιουργήσουμε το test dataset εφόσον δεν περνάμε στην μέθοδο ως όρισμα τα αντίστοιχα labels y τα οποία και χρειάζονται μαζί με τα Χ στην DigitsDataset;

**Comment by georgepar** [comment]
1. Θα χρησιμοποιήσετε την κλάση DigitsDataset για να φτιάξετε ένα object εντός της κλάσης.
2. Αυτό που κάνουμε συνήθως όταν δεν έχουμε διαθέσιμα labels στο test είναι να φτιάχνουμε 2 εκδοχές του dataset. Μια που γυρνάει τα labels για training / validation και μια που δε γυρνάει τα labels για testing.
